content/en/apps/tutorials/sms-forms.md

```diff
@@ -0,0 +1,413 @@
+
+## Implementation Steps
+
+SMS forms are defined using json in the `app_settings.json` file. Read [JSON forms]({{% ref "apps/reference/app-settings/forms#app_settingsjson-forms" %}}).
+
+### 1. Enable Transitions
+
+To ensure SMS forms function as expected, you will first need to able the following transitions in `app_settings.json`.
+
+```json
+  "update_clinics": true,
+  "registration": true,
+  "accept_patient_reports": true,
+  "generate_shortcode_on_contacts": true,
+  "death_reporting": true
+```
+
+{{< see-also page="apps/reference/app-settings/transitions" title="Transitions" >}}
+
+### 2. Define a Place Registration Form
+
+To define a form, edit the object corresponding to the `forms` key in `app_settings.json`.
+
+Add a household registration form by adding the code snippet below to the `forms` object.
+
+```json
+"HR": {
+  "meta": {
+    "code": "HR",
+    "label": {
+      "en": "New Household Registration"
+    }
+  },
+  "fields": {
+    "place_name": {
+      "labels": {
+        "tiny": {
+          "en": "household_name"
+        },
+        "description": {
+          "en": "Name of Household"
+        },
+        "short": {
+          "en": "Household name"
+        }
+      },
+      "position": 0,
+      "type": "string",
+      "length": [
+        3,
+        30
+      ],
+      "required": true
+    }
+  }
+}
+```
+
+{{% alert title="Note" %}} Users will register new households by sending a text message in the format `HR <household name>`. For example, `HR Mary` will register `Mary's Household`. {{% /alert %}}
+
+To set the validation rules and autoresponses, edit the array corresponding to the `registration` key in `app_settings.json`. Add the following object within the array.
+
+```json
+{
+  "form": "HR",
+  "events": [
+    {
+      "name": "on_create",
+      "trigger": "add_place",
+      "params": "{ \"contact_type\": \"clinic\" }",
+      "bool_expr": ""
+    }
+  ],
+  "validations": {
+    "join_respones": false,
+    "list": [
+      {
+        "property": "place_name",
+        "rule": "!regex('household') && !regex('house hold')",
+        "translation_key": "messages.hr.validation.no_household_in_name"
+      },
+      {
+        "property": "place_name",
+        "rule": "lenMin(1) && lenMax(30)",
+        "translation_key": "messages.hr.validation.household_name_length"
+      }
+    ]
+  },
+  "messages": [
+    {
+      "event_type":"report_accepted",
+      "recepient": "reporting_unit",
+      "translation_key": "messages.hr.report_accepted"
+    }
+  ]
+}
+```
+
+{{% alert title="Note" %}} You can also define your own [validation rules]({{< relref "apps/reference/app-settings#validations" >}}).  {{% /alert %}}
+
+{{% alert title="Note" %}} `translation_key` represents the message that is sent out. This will be defined in a translations file (covered in the translations tutorial). {{% /alert %}}
```

Comment:
We should avoid overuse of notes, and back-to-back notes. Can you bring one of these into the main text? Also, for consistency we should try to use `ref`, not `relref` in links.

Reply:
Updated and reduced the number of notes.